### PR TITLE
Fix BN doc

### PR DESCRIPTION
**Pull request by richsalz**
Backport from https://github.com/openssl/openssl/pull/5141

### CODE REVIEW

File: doc/crypto/BN_zero.pod [hunk 1]
```diff
@@ -25,19 +25,21 @@ Deprecated:
...
-BN_get_word() returns B<a>, if it can be represented as an unsigned
-long.
+BN_get_word() returns B<a>, if it can be represented as a B<BN_ULONG>.
 
 =head1 RETURN VALUES
 
-BN_get_word() returns the value B<a>, and 0xffffffffL if B<a> cannot
-be represented as an unsigned long.
+BN_get_word() returns the value B<a>, or all-bits-set if B<a> cannot
+be represented as a single integer.
```
Comment (bernd-edlinger) [inline doc/crypto/BN_zero.pod]: as a BN_ULONG ?

File: doc/crypto/BN_zero.pod [hunk 2]
```diff
@@ -46,10 +48,12 @@ The preferred version of BN_zero() never fails and returns no value.
 
 =head1 BUGS
 
-Someone might change the constant.
-
-If a B<BIGNUM> is equal to 0xffffffffL it can be represented as an
+If a B<BIGNUM> is equal to the value of all-bits-set, it will collide
 unsigned long but this value is also returned on error.
```
Comment (bernd-edlinger) [inline doc/crypto/BN_zero.pod]: remove this line,

File: doc/crypto/BN_zero.pod (hunk 1 above)
Comment (richsalz) [reply doc/crypto/BN_zero.pod]: yeah, i suppose.  but all three releases have the same wording.

**Comment by richsalz** [comment]
Updated commit pushed that addresses both comments.

File: doc/crypto/BN_zero.pod [hunk 3]
```diff
@@ -46,10 +48,11 @@ The preferred version of BN_zero() never fails and returns no value.
...
 =head1 BUGS
 
-Someone might change the constant.
+If a B<BIGNUM> is equal to the value of all-bits-set, it will collide
+with the error condition returned by BN_get_word() which uses that
+as an error value.
 
-If a B<BIGNUM> is equal to 0xffffffffL it can be represented as an
-unsigned long but this value is also returned on error.
+B<BN_ULONG> should probably be a typedef.
```
Comment (vdukhovni) [inline doc/crypto/BN_zero.pod]: I think the `should probably be a typedef` comment should go.  By now some feature tests likely exist that test for `ifdef BN_ULONG`, so it is too late to make this a type instead.  Also, when we add new types, applications that support older releases can test for macros more easily than for new types (instead they'd need to test against the OPENSSL_VERSION_NUMBER that adds the new type).

So all in all while may at some point introduce new types as non-macros, this one should probably stay as-is long-term.

**Comment by richsalz** [comment]
Merged, thanks!

**Pull request closed without merging**